src/PJ_ob_tran.c

```diff
@@ -89,76 +89,112 @@ static void *destructor(PJ *P, int errlev) {
         return pj_default_destructor (P, errlev);
 
     if (P->opaque->link)
-    {
-        /* This is a bit tricky: the linked PJ* shares the same params as */
-        /* the current one, so unset it to avoid double free */
-        /* We used to call P->opaque->link->pfree(P->opaque->link); only */
-        /* but this leaked grids */
-        P->opaque->link->params = NULL;
-        pj_free(P->opaque->link);
-    }
+        P->opaque->link->destructor (P->opaque->link, errlev);
 
     return pj_default_destructor(P, errlev);
 }
 
 
 
+
+/***********************************************************************
+
+These functions are modified versions of the functions "argc_params"
+and "argv_params" from PJ_pipeline.c
+
+Basically, they do the somewhat backwards stunt of turning the paralist
+representation of the +args back into the original +argv, +argc
+representation accepted by pj_init_ctx().
+
+This, however, also begs the question of whether we really need the
+paralist linked list representation, or if we could do with a simpler
+null-terminated argv style array? This would simplfy some code, and
+keep memory allocations more localized.
+
+***********************************************************************/
+
+typedef struct {int argc; char **argv;} ARGS;
+
+/* count the number of args in the linked list <params> */
+static size_t paralist_params_argc (paralist *params) {
+    size_t argc = 0;
+    for (; params != 0; params = params->next)
+        argc++;
+    return argc;
+}
+
+/* turn paralist into argc/argv style argument list */
+static ARGS ob_tran_target_params (paralist *params) {
+    char **argv;
+    size_t i = 0;
+    ARGS args = {0, 0};
+    size_t argc = paralist_params_argc (params);
+    if (0==argc)
+        return args;
+    args.argv = pj_calloc (argc, sizeof (char *));
```

Comment:
As far as I can see this args.argv is never deallocated

Reply:
Oops! Thanks! Correcting immediately